### ISSUE TITLE
[FEATURE REQUEST] Support Qt6

### ISSUE THREAD

**Issue by Neumann-A** [OpenImageIO/oiio#3759]
See patches in https://github.com/microsoft/vcpkg/pull/29078 to make it compile with Qt6

**Comment by mag-derrickcanfield** [comment]
I am unable to build OIIO via VCPKG due to an issue with QT6. I imagine this is related to the reason this feature request was requested; is that right? And if so, is there a known workaround for this?

Here is the VCPKG error
```
Building qtbase[brotli,concurrent,core,dbus,default-features,doubleconversion,freetype,gui,harfbuzz,icu,jpeg,network,opengl,openssl,pcre2,png,sql,sql-psql,sql-sqlite,testlib,thread,widgets,zstd]:x64-windows...
-- Using cached qtbase-everywhere-src-6.4.2.tar.xz.
-- Cleaning sources at D:/dev/git/vcpkg/buildtrees/qtbase/src/here-src-6-a23943cb65.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/dev/git/vcpkg/downloads/qtbase-everywhere-src-6.4.2.tar.xz
-- Applying patch allow_outside_prefix.patch
-- Applying patch clang-cl_source_location.patch
-- Applying patch config_install.patch
-- Applying patch fix_cmake_build.patch
-- Applying patch harfbuzz.patch
-- Applying patch fix_egl.patch
-- Applying patch fix_egl_2.patch
-- Applying patch clang-cl_QGADGET_fix.diff
-- Applying patch installed_dir.patch
-- Applying patch GLIB2-static.patch
-- Applying patch env.patch
-- Applying patch dont_force_cmakecache_latest.patch
-- Using source at D:/dev/git/vcpkg/buildtrees/qtbase/src/here-src-6-a23943cb65.clean
-- Found external ninja('1.11.0').
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
CMake Error at installed/x64-windows/share/vcpkg-cmake-config/vcpkg_cmake_config_fixup.cmake:58 (message):
  vcpkg_cmake_config_fixup was passed extra arguments:
  TOOLS_PATH;tools/Qt6/bin
Call Stack (most recent call first):
  ports/qtbase/cmake/qt_install_submodule.cmake:239 (vcpkg_cmake_config_fixup)
  ports/qtbase/cmake/qt_install_submodule.cmake:339 (qt_fixup_and_cleanup)
  ports/qtbase/portfile.cmake:272 (qt_install_submodule)
  scripts/ports.cmake:147 (include)


error: building qtbase:x64-windows failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
```

**Comment by Neumann-A** [comment]
> Here is the VCPKG error
> CMake Error at installed/x64-windows/share/vcpkg-cmake-config/vcpkg_cmake_config_fixup.cmake:58 (message):

how old is the installed tree of your vcpkg please? The message should come from line 7 now not 58. This is a vcpkg-cmake-config port which is over 1 year old.

**Comment by mag-derrickcanfield** [comment]
I pulled from the git repo as well as ran `vcpkg update` today prior to attempting to install. I imagine that is what you are asking about, but if not, what else should I be doing to make sure my VCPKG instance isn't severely out of date?

**Comment by Neumann-A** [comment]
> VCPKG instance isn't severely out of date?

Clean checkout, rebuild everything. `vcpkg update` is not really well supported and you probably missed the  `--no-dry-run`  option because otherwise the port should have been rebuild.

**Comment by Neumann-A** [comment]
Also `vcpkg upgrade --no-dry-run` is actually the command to update the ports in classic mode. `vcpkg update` just prints a message what could be updated.

**Comment by lgritz** [comment]
@Neumann-A Is this something where you can upstream the OIIO-side changes you made in the vcpkg patch? I'm just juggling too many things, and it would be very helpful if I could just approve a PR rather than have to reconstruct the right parts of the patch by hand. 

One potential difference is that it's important for us to still be able to build against Qt5 for those users (pretty much everybody in VFX) still using it, whereas the vcpkg side changes likely just assumed a wholesale upgrade to Qt6 without the need to retain the ability to build against either.

**Comment by mag-derrickcanfield** [comment]
That did the trick @Neumann-A ! Thanks for walking me through that. Hopefully I didn't pollute your thread too much with my nonsense.

**Comment by Neumann-A** [comment]
>  vcpkg side changes likely just assumed a wholesale upgrade to Qt6 without the need to retain the ability to build against either.

exactly. I just forced resolution to Qt6 without any regards for Qt5 which is why i cannot simply upstream the patch.

**Comment by lgritz** [comment]
Proposed fix: https://github.com/OpenImageIO/oiio/pull/3779

**Comment by lgritz** [comment]
Closing, this was fixed by #3779
